routes/adjustmentMethods/EToAdjustmentMethod.ts

```diff
@@ -0,0 +1,224 @@
+			radiation: Math.round( etoData.solarRadiation * 100) / 100
+		}
+	}
+}
+
+/* The implementation of this algorithm was guided by a step-by-step breakdown
+	(http://edis.ifas.ufl.edu/pdffiles/ae/ae45900.pdf) */
+/**
+ * Calculates the reference potential evapotranspiration using the Penman-Monteith (FAO-56) method
+ * (http://www.fao.org/3/X0490E/x0490e07.htm).
+ *
+ * @param etoData The data to calculate the ETo with.
+ * @param elevation The elevation above sea level of the watering site (in feet).
+ * @param coordinates The coordinates of the watering site.
+ * @return The reference potential evapotranspiration (in inches per day).
+ */
+export function calculateETo( etoData: EToData, elevation: number, coordinates: GeoCoordinates ): number {
+	// Convert to Celsius.
+	const minTemp = ( etoData.minTemp - 32 ) * 5 / 9;
+	const maxTemp = ( etoData.maxTemp - 32 ) * 5 / 9;
+	// Convert to meters.
+	elevation = elevation / 3.281;
+	// Convert to meters per second.
+	const windSpeed = etoData.windSpeed / 2.237;
+
+	const avgTemp = ( maxTemp + minTemp ) / 2;
+
+	const saturationVaporPressureCurveSlope = 4098 * 0.6108 * Math.exp( 17.27 * avgTemp / ( avgTemp + 237.3 ) ) / Math.pow( avgTemp + 237.3, 2 );
+
+	const pressure = 101.3 * Math.pow( ( 293 - 0.0065 * elevation ) / 293, 5.26 );
+
+	const psychrometricConstant = 0.000665 * pressure;
+
+	const deltaTerm = saturationVaporPressureCurveSlope / ( saturationVaporPressureCurveSlope + psychrometricConstant * ( 1 + 0.34 * windSpeed ) );
+
+	const psiTerm = psychrometricConstant / ( saturationVaporPressureCurveSlope + psychrometricConstant * ( 1 + 0.34 * windSpeed ) );
+
+	const tempTerm = ( 900 / ( avgTemp + 273 ) ) * windSpeed;
+
+	const minSaturationVaporPressure = 0.6108 * Math.exp( 17.27 * minTemp / ( minTemp + 237.3 ) );
+
+	const maxSaturationVaporPressure = 0.6108 * Math.exp( 17.27 * maxTemp / ( maxTemp + 237.3 ) );
+
+	const avgSaturationVaporPressure = ( minSaturationVaporPressure + maxSaturationVaporPressure ) / 2;
+
+	const actualVaporPressure = ( minSaturationVaporPressure * etoData.maxHumidity / 100 + maxSaturationVaporPressure * etoData.minHumidity / 100 ) / 2;
+
+	const dayOfYear = moment.unix( etoData.timestamp ).dayOfYear();
+
+	const inverseRelativeEarthSunDistance = 1 + 0.033 * Math.cos( 2 * Math.PI / 365 * dayOfYear );
+
+	const solarDeclination = 0.409 * Math.sin( 2 * Math.PI / 365 * dayOfYear - 1.39 );
+
+	const latitudeRads = Math.PI / 180 * coordinates[ 0 ];
+
+	const sunsetHourAngle = Math.acos( -Math.tan( latitudeRads ) * Math.tan( solarDeclination ) );
+
+	const extraterrestrialRadiation = 24 * 60 / Math.PI * 0.082 * inverseRelativeEarthSunDistance * ( sunsetHourAngle * Math.sin( latitudeRads ) * Math.sin( solarDeclination ) + Math.cos( latitudeRads ) * Math.cos( solarDeclination ) * Math.sin( sunsetHourAngle ) );
+
+	const clearSkyRadiation = ( 0.75 + 2e-5 * elevation ) * extraterrestrialRadiation;
+
+	const solarRadiation = etoData.solarRadiation;
```

Comment:
I think we should calculate the radiation in the WeatherProvider and include it in the EToData it returns. If you think the approximation method might also be useful to other WeatherProviders, we can add the function to EToAdjustmentMethod (along with `approximateSolarRadiation`, which can be renamed if necessary) so we don't have to repeat it in multiple WeatherProviders

Reply:
I think it would be a stretch for a PWS to claim it can report radiation if it doesn't have a radiation sensor. Whereas, I think it is fine for OWM/DarkSky to convert cloudCover to a radiation approximation. I have a sense that using cloudCover is a more accurate derivative than Temp but I could be very wrong on this. Appreciate any insight you have on this.

Reply:
I think that the local weather provider is a special case where we can use an "anything is better than nothing" policy when making approximations since it can only be used on servers the user is running locally. We'll create a documentation page explaining the different ways that any necessary approximations can be made, and try to list them in order of accuracy. If the user doesn't have a radiation or cloud coverage sensor but still wants to use their PWS as the weather provider for the EToAdjustmentMethod, it means that they've accepted that they want to calculate the watering scale using the limited data available, even if it won't be as accurate.

Reply:
Resolved in favor of Slack conversation